catalog/src/exceptions/transfer-execption.ts

```diff
@@ -0,0 +1,5 @@
+export class TransferError extends Error {
+  constructor(aggreementId: string) {
+    super(aggreementId)
+  }
+}
```

Comment:
maybe add a property with the agreementId to make it easier to fetch from the frontend
```suggestion
export class TransferError extends Error {
  public readonly agreementId: string
  constructor(message: string, aggreementId: string) {
    super(message)
    this.agreementId = aggreementId
  }
}
```